_posts/2019-04-24-server-rendering-responsively.md

```diff
@@ -0,0 +1,199 @@
+list of user-agents, we couldn't pass up the opportunity for more performance optimization. We have analytics about
+our users, and we know the browsers & devices we need to support. If we recognize your device, why not skip the
+non-mobile breakpoint when rendering from the server?
+
+We investigated existing user-agent detection libraries, but the ones we found didn't have the right data for us.
+We needed to know the minimum width for a browser on a given device, and if it was resizable, and to what
+dimensions it was resizable. If they _did_ have this data, they didn't provide it to us easily.
+
+So we did some experimentation, given the browsers & devices we knew we needed to support. And yeah...we
+(reluctantly) created our own user-agent detection library,
+[@artsy/detect-responsive-traits](https://github.com/artsy/detect-responsive-traits). This data is very specific to
+the browsers & devices we support on [artsy.net](artsy.net), but it's the data we need to optimize our responsive
+rendering. We're using it to determine if your browser is likely going to use only the mobile breakpoint of our
+app, in which case we don't have to also server-render the desktop version.
+
+TODO: is this sniffing part of react-responsive-media, or is it separate? I should find this out and make it clear.
+
+## The current state of @artsy/react-responsive-media
+
+With this, we now have:
+
+- TODO: list the features out.
+
+We're pretty happy with this, and we have generally good feelings about where it's at. We're proud of the fact that
+our approach works well for SSR, even in cases where you're doing _only_ SSR.
+
+## `styled-system`
+
+One other major factor in our approach to a responsive [artsy.net](artsy.net) is the
+[`styled-system`](https://styled-system.com/) project. We started using it around the same time we started building
+@artsy/react-responsive-media, and it has been incredibly helpful. While @artsy/react-responsive-media is great for
+rendering large sub-trees based on a matching breakpoint, `styled-system` is really great for styling individual
+components based on a matching breakpoint. For example, here's a component that would render a `div` (`Box`) with a
+width of 50% at our mobile breakpoint, and 100% for anything larger:
+
+```
+<Box width={["50%", "100%"]}>...</Box>
+```
+
+It's pretty amazing. `styled-system` allows us to use the same components at different breakpoints, but style them
+differently at each breakpoint. Without it, we were seeing _a lot_ of duplicated components via
+@artsy/react-responsive-media. With `styled-system`, we're finding that we use @artsy/react-responsive-media to
+occasionally handle significant layout differences across breakpoints, but `styled-system` more frequently for
+subtle styling differences across breakpoints.
+
+TODO: Have we already blogged about styled-system? I feel like we might have, and I should find out & link to that
+article.
```

Comment:
@zephraph is currently handling a post on Palette, which will likely go into styled-system. 

I'm happy this sub section is in - but it's really important to note that since Artsy as a team has started adopting responsive props I've seen zero new uses of RRM. This could be related to us fixing areas that we would naturally have to use it, but really responsive props are so flexible it's fairly easy to lay out complex markup designs and ensure that they work for every breakpoint using _only_ that mechanism. Maybe give it a bit more weight? I know from the very beginning this was the eventual aim -- lean on styled-system.

Reply:
It might also be good to point this out at the top of the post too. Some kind of "TLDR: Check out styled-system! https://styled-system.com/responsive-styles" (or something).